File: python/maximum_score_after_splitting_a_string.py
def score_split(s, i):
    return sum(1 for c in s[:i] if c == "0") \
        + sum(1 for c in s[i:] if c == "1")


class Solution:
    def maxScore(self, s: str) -> int:
        return max(score_split(s, i) for i in range(1, len(s)))

File: rust/maximum_score_after_splitting_a_string.rs
use std::cmp;

impl Solution {
    pub fn max_score(s: String) -> i32 {
        let total_ones: i32 = s.chars().filter(|&c| c == '1').count().try_into().unwrap();
        let mut current_zeroes = 0;
        let mut current_ones = total_ones;
        let mut result = 0;
        for (i, c) in s.chars().enumerate() {
            if i == s.len() - 1 {
                break;
            }
            if c == '1' {
                current_ones -= 1;
            } else {
                current_zeroes += 1;
            }
            result = cmp::max(result, current_ones + current_zeroes);
        }
        return result;
    }
}
